src/transform/adm0_surround.py

```diff
@@ -17,6 +17,101 @@
 
 ##
 #
+def get_neighbours_generic(gdf_source, gdf_aoi, uid):
+    bbox = gdf_aoi.bounds
+    minx = bbox.minx.iloc[0]
+    maxx = bbox.maxx.iloc[0]
+    miny = bbox.miny.iloc[0]
+    maxy = bbox.maxy.iloc[0]
+    # Ropey way to scale the WGS84 bounding box by a factor... 
+    # Should probably map these to see what the distortion ends up as.
+    factor = 0.1    # does this mean 10% ???
+    pcLon = abs(minx-maxx) / 360    # Total degrees X axis
+    lonDistScaled = 360 * ((pcLon * factor) + pcLon) 
+    pcLat = abs(miny-maxy) / 180    # Total degrees Y axis
+    latDistScaled = 180 * ((pcLat * factor) + pcLat)
+    # Use the centroid to extend out to scaled bbox
+    centX = minx + abs(minx-maxx)/2
+    centY = miny + abs(miny-maxy)/2
+    minxScale = centX - (lonDistScaled / 2)
+    maxxScale = centX + (lonDistScaled / 2)
+    minyScale = centY - (latDistScaled / 2)
+    maxyScale = centY + (latDistScaled / 2)
+    # Use CX method to query by scaled bounding box coords
+    print('Checking neighbour countries.')
+    print(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+    gdf_n_check = gdf_source.cx[minxScale:maxxScale, minyScale:maxyScale] 
+    print(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+    # Note the country list will also include the original country of
+    # interest.
+    return gdf_n_check[uid].unique().tolist()
+
+
+
+##
+#
+# Include the Transform functions for GeoBoundaries Surround here.
+# Principles will be the same. Get a World Dataset, pass an ISO code, 
+# query for surrounding intersects. Process all intersects.
+def transform_geoboundaries():
+    config = parse_yaml('config.yml')
+    rawdir = os.path.join(config['dirs']['raw_data'],
+                          config['geoboundaries']['subfolder'])
+    iso = config['constants']['ISO3']
+    isofield = config['geoboundaries']['isofield']
+    schema_filename =  os.path.join(
+            config['dirs']['schemas'], config['surrounding']['schema'])
+    output_filename = os.path.join(
+            config['dirs']['processed_data'], 
+            config['surrounding']['geoboundaries']['processed'])
+    dfs = []
+    for root, dirs, files in os.walk(rawdir):
+        for fileName in files:
+            if fileName.endswith(".zip"):
+                # Unzip
+                forUnzip, ext = os.path.splitext(fileName)
+                source = os.path.join(root, fileName)
+                print(f'Processing {source}')
+                try:
+                    geobndzip = zipfile.ZipFile(source, 'r')
+                except Exception as err:
+                    print(err)
```

Comment:
Some of these files return this error:
```File is not a zip file```
Any idea why?

Reply:
No not sure. When I checked manually I was unable to open the zips so just assume that some of the country download links aren't valid.

Reply:
Okay, maybe something to note on the confluence page then